File: platform_src/ios/elastOS/Classes/NotificationManager/NotificationRequest.swift

public class NotificationRequest {
    /** Identification key used to overwrite a previous notification if it has the same key. */
    public var key: String? = nil
    /** Title to be displayed as the main message on the notification. */
    public var title: String? = nil
    /** Detailed message for this notification. */
    public var message: String? = nil
    /** Intent URL emitted when the notification is clicked. */
    public var url: String? = nil
    /** Contact DID emitting this notification, in case of a remotely received notification. */
    public var emitter: String? = nil;

    public static func fromJSONObject(_ obj: Dictionary<String, Any>) -> NotificationRequest? {
        guard obj.keys.contains("key") && obj.keys.contains("title") else {
            return nil
        }

        let notif = NotificationRequest()

        notif.key = obj["key"] as? String
        notif.title = obj["title"] as? String
        
        if obj.keys.contains("message") {
            notif.message = obj["message"] as? String
        }

        if obj.keys.contains("url") {
            notif.url = obj["url"] as? String
        }

        if obj.keys.contains("emitter") {
            notif.emitter = obj["emitter"] as? String
        }

        return notif
    }
}

File: platform_src/electron/main/notificationmanager/NotificationRequest.ts
export class NotificationRequest {
    /** Identification key used to overwrite a previous notification if it has the same key. */
    public key: string = null;
    /** Title to be displayed as the main message on the notification. */
    public title: string = null;
    /** Intent URL emitted when the notification is clicked. */
    public url: string = null;
    /** Contact DID emitting this notification, in case of a remotely received notification. */
    public emitter: string = null;

    public static fromJSONObject(obj: any): NotificationRequest {

        let notif = new NotificationRequest();
        notif.key = obj.key as string;
        notif.title = obj.title as string;

        if (obj.url != null)
            notif.url = obj.url as string;
        if (obj.emitter != null) 
            notif.emitter = obj.emitter as string;
        return notif;
    }
}